lib/adapters/file-adapter.js

```diff
@@ -56,10 +72,28 @@ var Read = Juttle.proc.source.extend({
     start: function() {
         var self = this;
         var stream = this.fetch();
+        var last;
 
         self.parser.parseStream(stream, function(points) {
+            var validPoints = [];
             self.parseTime(points, self.timeField);
-            self.emit(points);
+            _.each(points, function(point) {
+                if (last && last.gt(point.time)) {
+                    // out of order points are dropped
+                    self.trigger('warning', self.runtime_error('RT-OUT-OF-ORDER'), {
+                        proc: 'read file'
+                    });
+                } else {
+                    // timeless points just pass through at this point
+                    if (point.time && point.time.moment) {
+                        if (point.time.gte(self.from) && point.time.lt(self.to)) {
```

Comment:
Wondering whether we want to emit an EOF as soon as we see a point thats more recent than `self.to`. This would be an optimization and not a semantic change in behavior so not necessary to address in this PR.